content/code_comments/introduction-to-code-commenting/_index.md

```diff
@@ -0,0 +1,247 @@
+          return n * factorial_recursive(n - 1)
+
+  ```
+
+  > In this example, the comment above the function provides a purposeful explanation of what the function does, clarifies the approach used (recursion), and warns about the potential issue of stack overflow for large input values.
+
+- Clarity and Readability: A good comment is clear, concise, and easy to read. It uses proper grammar, punctuation, and formatting conventions. Well-structured comments make it easier for developers to follow the code and comprehend its meaning.
+
+  ```python
+  # Function to calculate the sum of integers in a list
+  # Parameters:
+  # - numbers: A list of integers
+  # Returns:
+  # - The sum of all integers in the list
+  def calculate_sum(numbers):
+      sum = 0
+      for num in numbers:
+          sum += num
+      return sum
+  ```
+
+  > In this example, the comment above the function is well-structured and easy to read. It clearly describes the function's purpose, parameters, and return value. It also uses proper grammar and punctuation to ensure clarity and readability.
+
+- Relevant Information: A good comment provides relevant information that is not immediately apparent from the code itself. It adds value by offering insights, reasoning, or describing the intention behind a particular approach or solution.
+
+  ```python
+  # Calculate the total price of items in the shopping cart
+  total_price = 0
+
+  for item in shopping_cart:
+      subtotal = item.price * item.quantity
+      total_price += subtotal
+
+  # Apply a discount if the total price exceeds the spending_limit
+  if total_price > spending_limit:
+      discount = total_price * 0.1  # 10% discount for exceeding the spending_limit
+      total_price -= discount
+  ```
+
+  > In this example, the comment explains the purpose of the code, which is to calculate the total price of items in a shopping cart and apply a discount if the total price exceeds the spending limit. It clarifies the logic behind the code and provides insight into the discount calculation, which might not be immediately obvious from the code itself. The comment adds relevant information that helps other developers understand the intention and behaviour of the code segment.
+
+- Updates with Code Changes: A good comment is maintained and updated alongside the code. If the code undergoes modifications, the corresponding comments should be reviewed and revised to ensure they accurately reflect the updated logic.
+
+  ```javascript
+  // Function to calculate the factorial of a number using recursion
+  // This approach utilizes the mathematical definition of factorial
+  // and may cause a stack overflow for large input values.
+  function factorialRecursive(n) {
+    if (n === 0) {
+      return 1;
+    } else {
+      return n * factorialRecursive(n - 1);
+    }
+  }
+
+  // Updated version of the function to handle negative inputs as well
+  // This approach now includes a base case for negative numbers
```

Comment:
This doesn't seem very useful...

You threw away some of the information from the previous comment. Are you saying that the previous information was unnecessary? If it was unnecessary why was it there in the first place? 

Imagine a programmer who wants to make use of this function. What information do they need in order to use the function? Put that in the comment. 
It looks like the only thing they might need to know is that negative numbers aren't allkowed. The rest is pretty obvious just from reading this line: `function factorialRecursive(n)`